backend/src/api.py

```diff
@@ -92,6 +95,128 @@ class NodeData:
 T = TypeVar("T", bound=RunFn)
 
 
+def validateTypes(
+    wrapped_func: RunFn,
+    schema_id: str,
+    py_var: str,
+    py_type: Union[str, Any],
+    associated_type: Type,
+):
+    evaluated_py_type = py_type
+    if isinstance(py_type, str):
+        try:
+            # Allows us to use pipe unions still
+            if "|" in py_type:
+                py_type = f'Union[{py_type.replace(" |", ",")}]'
```

Comment:
I think there's an easier way to do this. https://stackoverflow.com/questions/50074956/how-exactly-do-eval-and-exec-interact-with-future

We could move this into its own file to make the connection between `from __future__ import annotations` and the type eval more explicit.

Reply:
I don't think this helps us. The current behavior seems to be that `eval` does not take `__future__` imports into account at all. All this StackOverflow says is that it _does_ take them into account (which is not the behavior I see) and how to disable it, which I don't need.

Reply:
Alright, looking at the docs, we could so this: 
```py
from __future__ import annotations

code_obj = compile(source_str, '<string>', 'eval', flags=annotations.compiler_flag)
eval(code_obj , globals, locals)
```

Reply:
that didn't do anything.